### PR TITLE
`Programming exercises`: Update Java and Kolin templates to use Ares 1.8.0

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [x] This is a small issue that I tested locally.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Changes affecting Programming Exercises
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context

Ares 1.8.0 got released and it requires additional changes to the pom.xml tempates. This PR incorporates them. It also replaces some uses of the outdated name with `Ares`.

### Description
- Update to Ares 1.8.0
- Add Maven Enforcer Plugin
- Update Documentation

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Programming Exercise

Make sure that:
- New Programming exercises can be created and tests can be executed using the default template
- Also build the documentation and make sure that `AJTS` is not mentioned anymore.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

**Review by MaisiKoleni — DISMISSED** [review]
Changes look good to me.

File: src/main/resources/templates/kotlin/test/projectTemplate/pom.xml [hunk 1]
```diff
@@ -102,6 +102,44 @@
...
+                    <rules>
+                        <requireFilesDontExist>
+                            <files>
+                                <file>${project.build.outputDirectory}/ch/qos/logback/</file>
+                                <file>${project.build.outputDirectory}/com/intellij/</file>
+                                <file>${project.build.outputDirectory}/com/sun/</file>
+                                <file>${project.build.outputDirectory}/de/tum/in/test/api/</file>
+                                <file>${project.build.outputDirectory}/java/</file>
+                                <file>${project.build.outputDirectory}/javax/</file>
+                                <file>${project.build.outputDirectory}/jdk/</file>
```
Comment (b-fein) [inline src/main/resources/templates/kotlin/test/projectTemplate/pom.xml:128]:
```suggestion
                                <file>${project.build.outputDirectory}/jdk/</file>
                                <file>${project.build.outputDirectory}/kotlin/</file>
```
This fixes the failing server tests.

**Review by b-fein — CHANGES_REQUESTED** [review]
Changes look good. Some server tests currently fail, I added an inline comment with a fix.

grep shows two more uses of AJTS/ajts, one in the documentation and one in a comment in the code:
```
src/main/java/de/tum/in/www1/artemis/repository/FeedbackRepository.java
203:            // overwrite AJTS: TimeoutException

docs/user/exercises/programming.rst
112:The examples and explanations are specifically written for Java (using AJTS/JUnit5), but the practices can also be generalized
```

File: src/main/resources/templates/kotlin/test/projectTemplate/pom.xml (hunk 1 above)
Comment (Strohgelaender) [reply src/main/resources/templates/kotlin/test/projectTemplate/pom.xml:128]: Thanks for pointing that out. 👍

**Review by MaisiKoleni — APPROVED** [review]
Sorry, I though all tests passed previously. Looks good to me.

**Pull request merged**